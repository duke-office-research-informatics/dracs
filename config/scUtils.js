import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
