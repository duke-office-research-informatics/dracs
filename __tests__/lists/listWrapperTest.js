import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "jest-axe";
import { List, SingleLineListItem } from "../../lib/dracs.es.js";

describe("List", () => {
  it("matches snapshot", () => {
    const list = mountWithTheme(<List />);
    expect(list).toMatchSnapshot();
  });
  it("returns no a11y violations", async () => {
    const list = mountWithTheme(<List />);
    const results = await axe(list.html());
    expect(results).toHaveNoViolations();
  });
  it("renders children", async () => {
    const list = mountWithTheme(
      <List>
        <SingleLineListItem title="its a test title" />
      </List>
    );
    expect(list).toMatchSnapshot();
    const results = await axe(list.html());
    expect(results).toHaveNoViolations();
  });
});
