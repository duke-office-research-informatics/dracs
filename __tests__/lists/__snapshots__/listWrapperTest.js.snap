// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot 1`] = `
.c0 {
  padding: 0;
  margin: 0;
}

<list__ListWrap>
  <ul
    className="c0"
  />
</list__ListWrap>
`;

exports[`List renders children 1`] = `
.c0 {
  padding: 0;
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
  list-style-type: none;
  width: 100%;
}

.c1:focus {
  outline: dotted 1px rgb(59,153,252);
  outline-offset: -1px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c3 {
  padding: 8px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<list__ListWrap>
  <ul
    className="c0"
  >
    <single_line__ItemWrap
      onKeyUp={[Function]}
      tabIndex={null}
    >
      <li
        className="c1"
        onKeyUp={[Function]}
        tabIndex={null}
      >
        <single_line__ItemLeftContainer>
          <div
            className="c2"
          >
            <single_line__TitleWrap>
              <div
                className="c3"
              >
                its a test title
              </div>
            </single_line__TitleWrap>
          </div>
        </single_line__ItemLeftContainer>
        <single_line__ItemRightContainer>
          <div
            className=""
          />
        </single_line__ItemRightContainer>
      </li>
    </single_line__ItemWrap>
  </ul>
</list__ListWrap>
`;
